test/testdrive/kafka-headers.td

```diff
@@ -0,0 +1,176 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ set keyschema={
+    "type": "record",
+    "name": "Key",
+    "fields": [
+        {"name": "key", "type": "string"}
+    ]
+  }
+
+$ set schema={
+        "type" : "record",
+        "name" : "test",
+        "fields" : [
+            {"name":"f1", "type":"string"},
+            {"name":"f2", "type":"long"}
+        ]
+    }
+
+$ kafka-create-topic topic=headers_src
+
+$ kafka-ingest format=avro topic=headers_src key-format=avro key-schema=${keyschema} schema=${schema} headers={"gus": "gusfive", "gus2": "gus3"}
+{"key": "fish"} {"f1": "fishval", "f2": 1000}
+
+$ kafka-ingest format=avro topic=headers_src key-format=avro key-schema=${keyschema} schema=${schema}
+{"key": "fish2"} {"f1": "fishval", "f2": 1000}
+
+
+> CREATE MATERIALIZED SOURCE headers_src
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC
+  'testdrive-headers_src-${testdrive.seed}'
+  KEY FORMAT AVRO USING SCHEMA '${keyschema}'
+  VALUE FORMAT AVRO USING SCHEMA '${schema}'
+  INCLUDE HEADERS
+  ENVELOPE UPSERT
+
+# empty case + has headers case
+> SELECT key, f1, f2, list_length(headers) from headers_src
+key     f1       f2     list_length
+-------------------------------------------
+fish    fishval  1000   2
+fish2   fishval  1000   0
```

Comment:
Wait, should this be `0`? Shouldn't this be null? Would be good to add a separate test for the empty header case.

Also if you do `headers::text` you'll be able to actually show the headers here, rather than just counting them.

Reply:
if you `INCLUDE HEADERS`, no headers == empty list of headers, not null. I think this is more clear, as there is no real distinction we are making, the cases are either no headers or some headers

Reply:
its a bit ugly to do `::text` cause its bytes not strings, but I can add this!

Reply:
Ohh, sorry, I see. rust-rdkafka incorrectly makes it seem like there is a difference between no headers and empty headers, but Kafka doesn't actually permit that.